File: PracticasProgramacion/Practica_08/cpp/IdentificadorTamanios.cpp
#include "IdentificadorTamanios.h"
#include <iostream>

void IdentificadorTamanios::mostrarTamaniosBasicos() const {
    std::cout << "--- Tipos basicos ---\n";
    std::cout << "sizeof(char)    = " << sizeof(char) << "\n";
    std::cout << "sizeof(int)     = " << sizeof(int) << "\n";
    std::cout << "sizeof(long)    = " << sizeof(long) << "\n";
    std::cout << "sizeof(float)   = " << sizeof(float) << "\n";
    std::cout << "sizeof(double)  = " << sizeof(double) << "\n";
    std::cout << "sizeof(bool)    = " << sizeof(bool) << "\n";
    std::cout << "sizeof(size_t)  = " << sizeof(size_t) << "\n";
}

void IdentificadorTamanios::mostrarTamaniosCreados() const {
    std::cout << "\n--- Tipos creados (PE) ---\n";
    std::cout << "sizeof(AutoPE)   = " << sizeof(AutoPE) << "\n";
    std::cout << "sizeof(PersonaPE)= " << sizeof(PersonaPE) << "\n";
    std::cout << "\n--- Tipos creados (POO) ---\n";
    std::cout << "sizeof(AutoPOO)   = " << sizeof(AutoPOO) << "\n";
    std::cout << "sizeof(PersonaPOO)= " << sizeof(PersonaPOO) << "\n";
}

File: PracticasProgramacion/Practica_08/python/identificador_tamanios.py
"""
IdentificadorTamanios: reporta sizeof equivalente para tipos básicos y creados.
En Python usamos sys.getsizeof; para clases/structs el tamaño es del objeto.
"""

import sys
from dataclasses import dataclass


@dataclass
class AutoPE:
    precio: float
    anio: int


@dataclass
class PersonaPE:
    nombre: str
    ap: str
    am: str
    genero: str
    edad: int


class AutoPOO:
    def __init__(self) -> None:
        self._precio = 0.0
        self._anio = 0


class PersonaPOO:
    def __init__(self) -> None:
        self._nombre = ""
        self._ap = ""
        self._am = ""
        self._genero = ""
        self._edad = 0


class IdentificadorTamanios:
    @staticmethod
    def mostrar_tamanios_basicos() -> None:
        print("--- Tipos basicos ---")
        for t, v in [
            ("int", 0),
            ("float", 0.0),
            ("bool", True),
            ("str (ej)", "x"),
        ]:
            print(f"sizeof({t}) ≈ {sys.getsizeof(v)}")

    @staticmethod
    def mostrar_tamanios_creados() -> None:
        print("\n--- Tipos creados (PE) ---")
        print(f"sizeof(AutoPE)   ≈ {sys.getsizeof(AutoPE(0.0, 0))}")
        print(f"sizeof(PersonaPE)≈ {sys.getsizeof(PersonaPE('','','','',0))}")
        print("\n--- Tipos creados (POO) ---")
        print(f"sizeof(AutoPOO)   ≈ {sys.getsizeof(AutoPOO())}")
        print(f"sizeof(PersonaPOO)≈ {sys.getsizeof(PersonaPOO())}")
